src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/AutoApplyStep.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Packages
+{
+    public abstract class AutoApplyStep : UpgradeStep
```

Comment:
Maybe call this AutoApplySubStep so that it's clear it's meant to be a base class for sub-steps rather than parents?